client/my-sites/plans/components/plans-header/index.tsx

```diff
@@ -64,17 +70,47 @@ const PlansHeader: React.FunctionComponent< {
 		subHeaderText ??
 		translate( 'See and compare the features available on each WordPress.com plan.' );
 
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const cart = useShoppingCart( selectedSiteId! );
+
 	if ( domainFromHomeUpsellFlow ) {
 		return <DomainUpsellHeader />;
 	}
 
+	async function handleAddCart() {
+		await cart.addProductsToCart( [
+			{
+				product_slug: PLAN_BUSINESS,
+				meta: 'agency-offer',
+			},
+			{
+				product_slug: PRODUCT_1GB_SPACE,
+				quantity: 50,
+				meta: 'agency-offer',
+			},
```

Comment:
I think we should add this depending on the user ID. We might offer 50GB for some agencies and 100GB for others depending on the deal.

Reply:
My idea of the `GET /sites/%s/special-offers` would support this. The endpoint could return different slugs for different users.